### PR TITLE
Added unit tests for daemon and options

### PR DESCRIPTION
**Pull request by duncanp-sonar**

### CODE REVIEW

File: src/Integration.Vsix.UnitTests/Settings/GeneralOptionsDialogControlTests.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+using Moq;
+using SonarLint.VisualStudio.Integration.Vsix;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Settings
+{
+    [TestClass]
+    public class GeneralOptionsDialogControlTests
+    {
+        [TestMethod]
+        public void Ctor_WithNullSettings_ThrowsArgumentNullException()
```
Comment (duncanp-sonar) [inline src/Integration.Vsix.UnitTests/Settings/GeneralOptionsDialogControlTests.cs]: Some of the new tests are pretty low-value i.e. just testing null arguments. However, these are for WPF components that are difficult to unit test.

File: src/TestInfrastructure/Helpers/MefTestHelpers.cs [hunk 2]
```diff
@@ -108,7 +108,10 @@ public static Export CreateExport<T>(object exportInstance, IDictionary<string,
             for (int i = 0; i < requiredExports.Count(); i++)
             {
                 // Try importing when not all of the required exports are available -> exception
-                Exceptions.Expect<Exception>(() => TryCompose(importer, catalog, requiredExports.Take(i)));
+                Exceptions.Expect<Exception>(
+                    null,
+                    () => TryCompose(importer, catalog, requiredExports.Take(i)),
+                    checkDerived: true);
```
Comment (duncanp-sonar) [inline src/TestInfrastructure/Helpers/MefTestHelpers.cs]: Fixed a bug in the existing test helper: the method doesn't care which exception is thrown so it's ok for it to be a sub-class of _Exception_ (_checkDerived=true_. _allowDerivedExceptions_ would have been a better name).

File: src/Integration.UnitTests/Helpers/SonarLintOutputTests.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            // Arrange
+            var spMock = new Mock<IServiceProvider>();
+            var spExport = MefTestHelpers.CreateExport<SVsServiceProvider>(spMock.Object);
+
+            // Act & Assert
+            MefTestHelpers.CheckTypeCanBeImported<SonarLintOutput, ISonarLintOutput>(null, new[] { spExport });
```
Comment (duncanp-sonar) [inline src/Integration.UnitTests/Helpers/SonarLintOutputTests.cs]: This (existing) helper method actually does quite a lot;
* it checks that the type can be imported
* it checks the type can't be imported unless all of the listed _required_ exports are supplied
* it checks the type can still be imported even if not all of the listed _optional_ exports are not supplied.

**Comment by duncanp-sonar** [comment]
No product changes, just adding unit test to get the coverage level up enough to turn the QG green.

File: src/Integration.UnitTests/Helpers/SonarLintOutputTests.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+            // Assert
+            act.ShouldThrow<ArgumentNullException>().And.ParamName.Should().Be("serviceProvider");
+        }
+
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            // Arrange
+            var spMock = new Mock<IServiceProvider>();
```
Comment (Evangelink) [inline src/Integration.UnitTests/Helpers/SonarLintOutputTests.cs]: [minor] I am not a big fan of short names it make it harder to read the code (you have to remember of the declaration).

File: src/Integration.Vsix.UnitTests/Settings/GeneralOptionsDialogPageTests.cs [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+            public void ApplyAccessor(ApplyKind applyBehavior)
+            {
+                base.OnApply(new PageApplyEventArgs { ApplyBehavior = applyBehavior });
+            }
+        }
+
+        [TestMethod]
+        public void OnActivate_ControlsAreConfiguredFromSettings1()
+        {
+            var settings = new ConfigurableSonarLintSettings()
```
Comment (Evangelink) [inline src/Integration.Vsix.UnitTests/Settings/GeneralOptionsDialogPageTests.cs]: [cosmetic] useless parenthesis when using inline initialization.

**Pull request merged**